pkg/internal/ebpf/watcher/watcher.go

```diff
@@ -0,0 +1,154 @@
+package watcher
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"io"
+	"log/slog"
+	"sync"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/ringbuf"
+
+	ebpfcommon "github.com/grafana/beyla/pkg/internal/ebpf/common"
+	"github.com/grafana/beyla/pkg/internal/exec"
+	"github.com/grafana/beyla/pkg/internal/goexec"
+	"github.com/grafana/beyla/pkg/internal/imetrics"
+	"github.com/grafana/beyla/pkg/internal/pipe"
+	"github.com/grafana/beyla/pkg/internal/request"
+	"github.com/grafana/beyla/pkg/internal/svc"
+)
+
+//go:generate $BPF2GO -cc $BPF_CLANG -cflags $BPF_CFLAGS -type watch_info_t -target amd64,arm64 bpf ../../../../bpf/watch_helper.c -- -I../../../../bpf/headers
+//go:generate $BPF2GO -cc $BPF_CLANG -cflags $BPF_CFLAGS -type watch_info_t -target amd64,arm64 bpf_debug ../../../../bpf/watch_helper.c -- -I../../../../bpf/headers -DBPF_DEBUG
+
+var newPortOfInterest = true // set to true initially to let port scanning run once at least
+var mux = sync.Mutex{}
+
+type BPFWatchInfo bpfWatchInfoT
+
+type Tracer struct {
```

Comment:
Since this has a different functionality than the Process Tracers, I think that you don't have to implement the `ebpf.Tracer` interface, and just use it directly from the watcher.